.github/workflows/verify-pacts-main.yml

```diff
@@ -0,0 +1,79 @@
+name: Verify Pacts on Main
+
+on:
+  push:
+    branches: [ main ]
+    paths-ignore: [ '*.md' ]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'gradle'
+
+      - name: Build all projects without running tests
+        run: ./gradlew --build-cache build -x test
+
+  verify-pacts:
+    needs: [ build ]
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
```

Comment:
I don't think postgres should be needed, but I'm leaving it in for now because we will need to test this action after this PR gets merged. In a nutshell, the reason it has to be merged first is that the GitHub API won't let us trigger this action on *any* branch until a version of it is merged into `main`. So once we merge this, we can iterate in a new PR.